src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/spark/SparkExecutor.scala

```diff
@@ -72,10 +72,11 @@ object SparkExecutor {
   private def getUpdatedJobInstance(
     jobInstance: JobInstance,
     app: App
-  ): JobInstance = {
+  )(implicit sparkConfig: SparkConfig): JobInstance = {
     val diagnostics = app.diagnostics match {
-      case "" => None
-      case _  => Some(app.diagnostics)
+      case ""                                => None
+      case _ if !sparkConfig.saveDiagnostics => None
```

Comment:
I would formulate that positively.
And "default" case left with None  always fallback to None.

`    val diagnostics = app.diagnostics match {
      case ""                                => None
      case _ if sparkConfig.saveDiagnostics  => Some(app.diagnostics)
      case _                                 => None
    }`